lib/src/linear_feedback_shift_register.dart

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) 2023 Intel Corporation
+
+// SPDX-License-Identifier: BSD-3-Clause
+
+//
+
+// linear_feedback_shift_register.dart
+
+// Implementation of Galois Linear Feedback Shift Register.
+
+//
+
+// 2024 October 1
+
+// Author: Omonefe Itietie <omonefe.itietie@intel.com>
+
+//
+
+import 'dart:collection';
+import 'package:rohd/rohd.dart';
+
+/// Galois Linear Feedback Shift Register
+class LinearFeedbackShiftRegister extends Module {
+  /// Contains polynomial size for LFSR
+  final int width;
+
+  /// Contains seed of LFSR (starting value)
+  Logic state;
+
+  /// Data names for signals
+  final String dataName;
+
+  /// Contains bit string that will be used to calculate the output
+  final Logic taps;
+
+  /// Output for shift register
+  Logic get dataOut => output('${dataName}_out');
+
+  /// The number of stages in this shift register.
+  final int shifts;
+
+  /// A [List] of [output]s where the `n`'th entry corresponds to a version of
+  /// the input data after passing through `n + 1` flops.
+  late final List<Logic> stages = UnmodifiableListView(
+      [for (var i = 0; i < shifts; i++) output(_stageName(i))]);
+
+  /// The name of the signal (and output pin) for the [i]th stage.
+  String _stageName(int i) => '${dataName}_stage_$i';
+
+  LinearFeedbackShiftRegister(Logic dataIn,
+      {required Logic clk,
+      required this.state,
+      required this.shifts,
+      required this.taps,
+      Logic? enable,
+      Logic? reset,
+      dynamic resetValue,
+      this.dataName = 'data'})
+      : width = dataIn.width,
+        super(name: '${dataName}_lfsr') {
+    dataIn = addInput('${dataName}_in', dataIn, width: width);
+    clk = addInput('clk', clk);
+    addOutput('${dataName}_out', width: width);
+
+    Map<Logic, dynamic>? resetValues;
+
+    if (reset != null) {
+      reset = addInput('reset', reset);
+      if (resetValue != null) {
```

Comment:
the original shift register was recently upgraded to allow for a `List` `resetValue` to control each element of the register (https://github.com/intel/rohd-hcl/blob/main/lib/src/shift_register.dart).  However, since the LFSR is always 1-wide, maybe a different API makes sense.  The current `resetValue` just allows you to basically set all flops to 0 or 1.  Should we allow a bitvector of width == `shifts` to initialize the LFSR?